doc/decisions/1_in_discussion/spec_expressiveness.md

```diff
@@ -2,22 +2,155 @@
 
 ## Problem
 
-Currently, you can easily come into wrong assumptions that something would work in the specification.
+Specifications can help users define certain details about keys and make sure only valid keys are used.
 
-We need to find minimal requirements to implement a sane spec plugin.
+One example of such as specification can be found below:
+
+```ni
+[public/ip]
+meta:/default = 127.0.0.1
+meta:/description = "The IP via which this application is accessible."
+
+[public/port]
+meta:/default = 8080
+meta:/description = "The port on which this application is accessible."
+```
+
+Here one can see `default` and `description` used as metadata. 
+
+The current implementation of the spec plugin is copying this metadata to all other namespaces.
+It also assures that if keys do not exist and have `default` metadata, it creates a cascading key.
+
+Assume we have a REST application we want to write a specification for.
+This REST application has to use multiple services.
```

Comment:
What is this first example for?

```suggestion
```

Reply:
I started this decision to first explain how something works and then slowly going towards to actual problem so everyone understands why this problem exists.